### PR TITLE
Allow rollbacks in FlexSearch

### LINKED ISSUE #3230: Add functionality for rolling back MarketCreated events from FlexSearch
https://github.com/AugurProject/augur/blob/a0a937a2014424a5db6f524663d19fb5f0e32e25/packages/augur-sdk/src/state/db/SyncableFlexSearch.ts#L113-L114

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a `TODO` comment in a0a937a2014424a5db6f524663d19fb5f0e32e25. It&#x27;s been assigned to @adrake33 because they committed the code.

This needs to fix a console error that was introduced when FlexSearch was moved to its own syncable class.

This should also add the new extraInfo property to the flexsearch index and enable workers + cache + async options in flexsearch.

### PR DESCRIPTION
**Pull request by adrake33**
Closes #3230

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
This pull request has been automatically deployed to [FeaturePeek](https://featurepeek.com). 🥇
 To view it, be sure to [join Augur Project](https://dashboard.featurepeek.com/login) on FeaturePeek.

 This link will always point to this PR's latest changes: https://peek.run/r9zh7p7

File: packages/augur-sdk/src/state/db/SyncableDB.ts [hunk 1]
```diff
@@ -109,7 +124,7 @@ export class SyncableDB extends AbstractDB {
       return -1;
     }
 
-    if (this.eventName === "OrderEvent") {
+    if (this.eventName === 'OrderEvent') {
```
Comment (bthaile) [inline packages/augur-sdk/src/state/db/SyncableDB.ts:127]: is there constants for eventNames?

File: packages/augur-sdk/src/state/db/SyncableDB.ts [hunk 2]
```diff
@@ -176,6 +191,18 @@ export class SyncableDB extends AbstractDB {
       for (const doc of blocksToRemove.docs) {
         await this.db.remove(doc._id, doc._rev);
       }
+
+      // Remove MarketCreated docs to FlexSearch in web worker (so that unit tests will run)
+      if (this.eventName === 'MarketCreated') {
```
Comment (bthaile) [inline packages/augur-sdk/src/state/db/SyncableDB.ts:196]: is there constants for eventNames to compare against?

File: packages/augur-ui/src/modules/events/actions/log-handlers.ts [hunk 3]
```diff
@@ -120,6 +121,14 @@ export const handleNewBlockLog = (log: Events.NewBlock) => (
   if (getState().authStatus.isLogged) dispatch(updateAssets());
 };
 
+export const handleMarketCreatedBulkSyncFinished = (log: any) => {
+  Augur.syncableFlexSearch.addMarketCreatedDocs(log.marketCreatedDocs);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/events/actions/log-handlers.ts:125]: I don't think this should be handled in augur-ui.

File: packages/augur-ui/src/modules/events/actions/log-handlers.ts [hunk 4]
```diff
@@ -120,6 +121,14 @@ export const handleNewBlockLog = (log: Events.NewBlock) => (
...
+};
+
+export const handleMarketCreatedRollbackFinished = (log: any) => {
+  Augur.syncableFlexSearch.removeMarketCreatedDocs(log.marketCreatedDocs);
```
Comment (bthaile) [inline packages/augur-ui/src/modules/events/actions/log-handlers.ts:129]: I don't think this should be handled in augur-ui

**Review by bthaile — CHANGES_REQUESTED** [review]
some comments and I don't think there should be changes in augur-ui for handling flex search, if I'm understanding it correctly.

**Review by bthaile — APPROVED** [review]
approving as a temporary measure for getting working

**Pull request merged**